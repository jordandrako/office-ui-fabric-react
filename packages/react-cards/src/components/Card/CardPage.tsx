import * as React from 'react';
import { ExampleCard, IComponentDemoPageProps, ComponentPage, PageMarkdown, PropertiesTableSet } from '@uifabric/example-app-base';

import { CardBasicExample } from './examples/Card.Basic.Example';
const CardBasicExampleCode = require('!raw-loader!@uifabric/react-cards/src/components/Card/examples/Card.Basic.Example.tsx') as string;

import { CardCompactExample } from './examples/Card.Compact.Example';
const CardCompactExampleCode = require('!raw-loader!@uifabric/react-cards/src/components/Card/examples/Card.Compact.Example.tsx') as string;

export class CardPage extends React.Component<IComponentDemoPageProps, {}> {
  public render(): JSX.Element {
    return (
      <ComponentPage
        title="Card"
        componentName="Card"
        exampleCards={
          <div>
            <ExampleCard title="Basic Card" code={CardBasicExampleCode}>
              <CardBasicExample />
            </ExampleCard>
            <ExampleCard title="Compact Card" code={CardCompactExampleCode}>
              <CardCompactExample />
            </ExampleCard>
          </div>
        }
        propertiesTables={
          <PropertiesTableSet sources={[require<string>('!raw-loader!@uifabric/react-cards/src/components/Card/Card.types.ts')]} />
        }
        overview={
          <PageMarkdown>{require<string>('!raw-loader!@uifabric/react-cards/src/components/Card/docs/CardOverview.md')}</PageMarkdown>
        }
        bestPractices={<div />}
        dos={<PageMarkdown>{require<string>('!raw-loader!@uifabric/react-cards/src/components/Card/docs/CardDos.md')}</PageMarkdown>}
        donts={<PageMarkdown>{require<string>('!raw-loader!@uifabric/react-cards/src/components/Card/docs/CardDonts.md')}</PageMarkdown>}
        isHeaderVisible={this.props.isHeaderVisible}
      />
    );
  }
}
