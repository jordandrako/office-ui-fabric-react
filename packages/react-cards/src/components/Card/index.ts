export * from './CardItem/CardItem';
export * from './Card';
export * from './Card.types';
