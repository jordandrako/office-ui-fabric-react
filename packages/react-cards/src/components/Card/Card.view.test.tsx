import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { CardView } from './Card.view';

// Views are just pure functions with no statefulness, which means they can get full code coverage
//    with snapshot tests exercising permutations of the props.
describe('CardView', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<CardView />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
