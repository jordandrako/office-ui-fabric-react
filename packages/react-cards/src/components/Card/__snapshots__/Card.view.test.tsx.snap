// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardView renders correctly 1`] = `
<div
  className=
      ms-Stack
      {
        align-items: space-between;
        box-sizing: border-box;
        display: flex;
        flex-direction: column;
        flex-wrap: nowrap;
        height: 100%;
        justify-content: space-between;
        width: auto;
      }
      & > * {
        text-overflow: ellipsis;
      }
      & > *:not(:first-child) {
        margin-top: 0px;
      }
      & > *:not(.ms-StackItem) {
        flex-shrink: 1;
      }
/>
`;
